woocommerce/woocommerce-ios
# Store onboarding: Dismiss payment setup web view when setup completes

Closes: #10573 


## Description

This PR updates the handling for payment setup web view to detect the completion and dismiss the web view when appropriate.
- Add a new view model for the setup web view with a callback closure for handling completion.
- Check for the params `wcpay-connection-error` and `wcpay-connection-success` to trigger the completion closure.

## Testing instructions

- Create a new JN site with Jetpack and Woo. Set up your site with a WCPay-supported country (e.g. US).
- Follow the setup steps in P91TBi-4BH-p2 up to the step of installing the dev plugin only.
- Log in to the app and switch to the JN site.
- On the My Store screen, select Get Paid on the onboarding task list.
- Select Continue Setup > Finish Setup.
- Tap the link to go back to WooPayment, notice that the web view is dismissed.
- Tap the Get Paid task again, this time finish the setup in test mode. When the setup completes (the page says "Redirecting to WooPayments..."), the web view should be dismissed.

## Screenshots

When canceling setup:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/f34c1d7d-679e-4e23-91b6-1cbb6aab5e12

When finishing setup:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/8a4cce58-c4f8-4223-ab50-0586f7f09fda


---
